lib/pmdtester/builders/project_hasher.rb

```diff
@@ -117,15 +117,24 @@ def make_violation_hash(file_ref, violation, is_diff = TRUE)
         'l' => violation.line,
         'f' => file_ref,
         'r' => violation.rule_name,
-        'm' => is_diff && violation.changed? ? diff_fragments(violation) : violation.message
+        'm' => create_violation_message(violation, is_diff)
       }
       h['ol'] = violation.old_line if is_diff && violation.changed? && violation.line != violation.old_line
       h
     end
 
-    def diff_fragments(violation)
-      diff = Differ.diff_by_word(violation.message, violation.old_message)
+    def create_violation_message(violation, is_diff)
+      return escape_html(violation.message) unless is_diff
```

Comment:
the old code checked not only against `is_diff`, but also for `violation.changed?`… is this change intentional?

Reply:
Good catch! I think we need to check for both (although I didn't see a difference in behavior yet). Only when violation.changed? is true, we would have a violation.old_message available. Maybe that explains, why I needed line 135 to deal with nil...